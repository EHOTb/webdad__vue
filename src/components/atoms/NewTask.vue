<template>
    <div :class="[$style.item, $style.new]">
                <input :class="$style.input" type="text" name="text" id="newTask" placeholder="Add new task" />
            </div>
</template>

<script>
export default {
    name: "NewTask"
}
</script>

<style lang="scss" module>
@import "../../assets/styles/var.scss";

.item {
    background: $bg-color-light;
    display: flex;
    justify-content: space-between;
    width: 80%;
    padding: 1rem;
    border-radius: 1rem;
    margin: 1.2rem;
}
.new {
    display: block;
    text-align: center;
    background-color: $bg-color-maxLight;
    color: $bg-color-dark;
    border: 0.2rem dashed $bg-color-dark;
}

.input {
        display: block;
        width: 100%;
        font-family: inherit;
        font-size: 1rem;
        font-weight: 400;
        line-height: 1.5;
        color: #212529;
        background-color: $bg-color-maxLight;
        border: none;
    }

    .input::placeholder {
        text-align: center;
        color: $bg-color-dark;
        font-size: 1.25rem;
    }
</style>

