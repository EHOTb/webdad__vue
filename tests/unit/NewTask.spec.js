import { mount } from "@vue/test-utils";
import NewTask from "../../src/components/atoms/NewTask";

describe("Компонент NewTask", () => {
  it("Snapshot NewTask", () => {
    const wrapper = mount(NewTask);
    expect(wrapper.element).toMatchSnapshot();
  });
  it("emit события NewTask", () => {
    const wrapper = mount(NewTask);
    wrapper.find("input").trigger("enter");
    wrapper.find("input").trigger("change");
    expect(wrapper.emitted().createNewTask).toBeTruthy();
  });
  //   it("props", () => {
  //     const wrapper = mount(NewTask, {
  //       propsData: {
  //         value: "value",
  //       },
  //     });
  //     expect(wrapper.find("input").element.value).toEqual( "value");
  //   });
});
