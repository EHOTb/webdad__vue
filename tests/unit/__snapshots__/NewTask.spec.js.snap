// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Компонент NewTask Snapshot NewTask 1`] = `
<div
  class="wrapper"
>
  <input
    class="input"
    name="text"
    placeholder="Add new task"
    type="text"
  />
</div>
`;
